### PR TITLE
VR-5199 Client: ModelVersion: associate requirements

### PR DESCRIPTION
**Pull request by nhatsmrt**

### CODE REVIEW

File: client/verta/verta/_registry/modelversion.py [hunk 1]
```diff
@@ -89,12 +94,15 @@ def get_asset(self, key):
     def del_asset(self, key):
         raise NotImplementedError
 
-    def set_environment(self, env):
-        # Env must be an EnvironmentBlob. Let's re-use the functionality from there
-        raise NotImplementedError
+    def log_environment(self, env):
+        self._refresh_cache()
+        self._msg.environment.CopyFrom(env._msg)
+        self._update()
```
Comment (convoliution) [inline client/verta/verta/_registry/modelversion.py]: Let's add a typecheck on `env`. I think it'll be
```python
if not isinstance(env, python._environment._Environment):
    raise TypeError("`env` must be of type Environment, not {}".format(type(env))

```

**Review by convoliution — CHANGES_REQUESTED** [review]

File: client/verta/verta/_registry/modelversion.py (hunk 1 above)
Comment (nhatsmrt) [reply client/verta/verta/_registry/modelversion.py]: I believe this is what you meant?
```python
from ..environment import _Environment
...
if not isinstance(env, _Environment):
        raise TypeError("`env` must be of type Environment, not {}".format(type(env)))
```

Comment (nhatsmrt) [reply client/verta/verta/_registry/modelversion.py]: Fixed.

Comment (convoliution) [reply client/verta/verta/_registry/modelversion.py]: That is indeed cleaner, thanks

**Review by convoliution — APPROVED** [review]

**Pull request merged**